SonarQube.Client/Requests/RequestFactory.cs

```diff
@@ -98,10 +99,16 @@ public RequestFactory(ILogger logger)
         /// SonarQube version to return a request implementation for. When the provided value is null
         /// returns the registered implementation with the highest version number.
         /// </param>
+        /// <param name="service">The current SonarQubeService instance</param>
         /// <returns>New the newest TRequest implementation for the specified SonarQube version.</returns>
-        public TRequest Create<TRequest>(Version version)
+        public TRequest Create<TRequest>(Version version, ISonarQubeService service)
             where TRequest : IRequest
         {
+            if (service == null)
```

Comment:
Note: it's ok for the version to be null (the `param` doc comment above says that passing a null version selects the latest available version)